src/Microsoft.DotNet.ProjectJsonMigration/Microsoft.DotNet.Internal.ProjectModel/ProjectReader.cs

```diff
@@ -850,11 +863,132 @@ private static void AddProjectFilesCollectionDiagnostics(JObject rawProject, Pro
                     ErrorCodes.DOTNET1015,
                     $"The '{option}' option is deprecated. Use {message} instead.",
                     project.ProjectFilePath,
-                    DiagnosticMessageSeverity.Warning,
+                    severity,
                     lineInfo.LineNumber,
                     lineInfo.LinePosition));
         }
 
+        private static void ConvertDeprecatedToSupportedFormat(JObject rawProject)
```

Comment:
I would normally ask for a refactoring of these methods. They seem extremely repeated and feel like they should be in their own abstraction (s). But, given the limited return we are going to get from it, I think it is fine to leave it for now. If we start making more and more changes here, then maybe we should extract the abstractions.